servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DefaultDnsClient.java

```diff
@@ -195,8 +198,7 @@ final class DefaultDnsClient implements DnsClient {
 
     @Override
     public String toString() {
-        return getClass().getSimpleName() + // FIXME: change to the name/id when builder requires it
-                '@' + toHexString(hashCode());
+        return id;
```

Comment:
`id` is the best effort, there is still a risk for having same ids for multiple instances. In RRLB we also append `hashCode` for all logs to make sure we can distinguish them:

https://github.com/apple/servicetalk/blob/469dc27317e6e7d83950100d060577bb2a8e7587/servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java#L156

```suggestion
        return id + " (instance @" + toHexString(hashCode()) + ')';
```